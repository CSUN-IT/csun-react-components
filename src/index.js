import MetaPrimaryNav from './components/meta/Nav';
import MetaBreadcrumbs from './components/meta/Breadcrumbs';
import MetaSubNav from './components/meta/SubNav';
import MetaButton from './components/meta/Button';
import MetaContainer from './components/meta/Container';
import MetaRow from './components/meta/Row';
import MetaColumn from './components/meta/Column';
import MetaFormField from './components/meta/FormField';
import MetaFormGroup from './components/meta/FormGroup';
import MetaTag from './components/meta/Tag';
import MetaPanel from './components/meta/Panel';
import MetaAlert from './components/meta/Alert';
import MetaList from './components/meta/List';
import MetaSideNav from './components/meta/SideNav';
import MetaAccordion from './components/meta/Accordion';
import MetaChatBubble from './components/web/ChatBubble';
import WebPrimaryNav from './components/web/PrimaryNav';
import WebHeader from './components/web/Header';
import WebSidebar from './components/web/Sidebar';

export {
  MetaBreadcrumbs,
  MetaPrimaryNav,
  MetaSubNav,
  MetaButton,
  MetaContainer,
  MetaRow,
  MetaColumn,
  MetaTag,
  MetaPanel,
  MetaFormField,
  MetaFormGroup,
  MetaAlert,
  MetaList,
  MetaSideNav,
  MetaAccordion,
  MetaChatBubble,
  WebPrimaryNav,
  WebHeader,
  WebSidebar,
};
