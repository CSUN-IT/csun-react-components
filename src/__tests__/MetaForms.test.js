import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { MetaFormField, MetaFormGroup } from '../../src';

configure({ adapter: new Adapter() });

describe('MetaFormGroup', () => {
  it('Should be defined', () => {
    expect(MetaFormGroup).toBeDefined();
  });

  it('Should be defined', () => {
    expect(MetaFormField).toBeDefined();
  });

  it('Renders component correctly', () => {
    const mockFn = jest.fn();
    const tree = shallow(
      <MetaFormGroup onSubmit={mockFn}>
        <MetaFormField
          field={{
            inputType: 'input',
            id: 'inputFied',
            placeholder: 'Type a message...',
            type: 'text',
            autoComplete: 'off',
            ariaLabel: 'input field',
          }}
          value={''}
        />
        <MetaFormField
          field={{
            inputType: 'textarea',
            id: 'textFied',
            placeholder: 'Type a message...',
            type: 'text',
            autoComplete: 'off',
          }}
          value={''}
        />
        <br />
        <MetaFormField
          field={{
            inputType: 'select',
            id: 'selectField',
            type: 'checkbox',
            selectOptions: [
              {
                value: '1',
                text: 'Example 1',
              },
              {
                value: '2',
                text: 'Example 2',
              },
            ],
          }}
          value={''}
        />
        <br />
        <MetaFormField
          field={{
            inputType: 'datepicker',
            id: 'datepicker',
            type: 'date',
          }}
          value={''}
        />
      </MetaFormGroup>,
    );
    expect(tree).toMatchSnapshot();
  });
});
