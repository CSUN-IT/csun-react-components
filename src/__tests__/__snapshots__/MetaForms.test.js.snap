// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaFormGroup Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaFormGroup
    className=""
    onSubmit={[MockFunction]}
    submitFunction={[Function]}
  >
    <MetaFormField
      changeFunction={[Function]}
      className=""
      field={
        Object {
          "ariaLabel": "input field",
          "autoComplete": "off",
          "id": "inputFied",
          "inputType": "input",
          "placeholder": "Type a message...",
          "type": "text",
        }
      }
      value=""
    />
    <MetaFormField
      changeFunction={[Function]}
      className=""
      field={
        Object {
          "autoComplete": "off",
          "id": "textFied",
          "inputType": "textarea",
          "placeholder": "Type a message...",
          "type": "text",
        }
      }
      value=""
    />
    <br />
    <MetaFormField
      changeFunction={[Function]}
      className=""
      field={
        Object {
          "id": "selectField",
          "inputType": "select",
          "selectOptions": Array [
            Object {
              "text": "Example 1",
              "value": "1",
            },
            Object {
              "text": "Example 2",
              "value": "2",
            },
          ],
          "type": "checkbox",
        }
      }
      value=""
    />
    <br />
    <MetaFormField
      changeFunction={[Function]}
      className=""
      field={
        Object {
          "id": "datepicker",
          "inputType": "datepicker",
          "type": "date",
        }
      }
      value=""
    />
  </MetaFormGroup>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MetaFormField
          changeFunction={[Function]}
          className=""
          field={
            Object {
              "ariaLabel": "input field",
              "autoComplete": "off",
              "id": "inputFied",
              "inputType": "input",
              "placeholder": "Type a message...",
              "type": "text",
            }
          }
          value=""
        />,
        <MetaFormField
          changeFunction={[Function]}
          className=""
          field={
            Object {
              "autoComplete": "off",
              "id": "textFied",
              "inputType": "textarea",
              "placeholder": "Type a message...",
              "type": "text",
            }
          }
          value=""
        />,
        <br />,
        <MetaFormField
          changeFunction={[Function]}
          className=""
          field={
            Object {
              "id": "selectField",
              "inputType": "select",
              "selectOptions": Array [
                Object {
                  "text": "Example 1",
                  "value": "1",
                },
                Object {
                  "text": "Example 2",
                  "value": "2",
                },
              ],
              "type": "checkbox",
            }
          }
          value=""
        />,
        <br />,
        <MetaFormField
          changeFunction={[Function]}
          className=""
          field={
            Object {
              "id": "datepicker",
              "inputType": "datepicker",
              "type": "date",
            }
          }
          value=""
        />,
      ],
      "className": " form__group",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "changeFunction": [Function],
          "className": "",
          "field": Object {
            "ariaLabel": "input field",
            "autoComplete": "off",
            "id": "inputFied",
            "inputType": "input",
            "placeholder": "Type a message...",
            "type": "text",
          },
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "changeFunction": [Function],
          "className": "",
          "field": Object {
            "autoComplete": "off",
            "id": "textFied",
            "inputType": "textarea",
            "placeholder": "Type a message...",
            "type": "text",
          },
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "changeFunction": [Function],
          "className": "",
          "field": Object {
            "id": "selectField",
            "inputType": "select",
            "selectOptions": Array [
              Object {
                "text": "Example 1",
                "value": "1",
              },
              Object {
                "text": "Example 2",
                "value": "2",
              },
            ],
            "type": "checkbox",
          },
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "changeFunction": [Function],
          "className": "",
          "field": Object {
            "id": "datepicker",
            "inputType": "datepicker",
            "type": "date",
          },
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MetaFormField
            changeFunction={[Function]}
            className=""
            field={
              Object {
                "ariaLabel": "input field",
                "autoComplete": "off",
                "id": "inputFied",
                "inputType": "input",
                "placeholder": "Type a message...",
                "type": "text",
              }
            }
            value=""
          />,
          <MetaFormField
            changeFunction={[Function]}
            className=""
            field={
              Object {
                "autoComplete": "off",
                "id": "textFied",
                "inputType": "textarea",
                "placeholder": "Type a message...",
                "type": "text",
              }
            }
            value=""
          />,
          <br />,
          <MetaFormField
            changeFunction={[Function]}
            className=""
            field={
              Object {
                "id": "selectField",
                "inputType": "select",
                "selectOptions": Array [
                  Object {
                    "text": "Example 1",
                    "value": "1",
                  },
                  Object {
                    "text": "Example 2",
                    "value": "2",
                  },
                ],
                "type": "checkbox",
              }
            }
            value=""
          />,
          <br />,
          <MetaFormField
            changeFunction={[Function]}
            className=""
            field={
              Object {
                "id": "datepicker",
                "inputType": "datepicker",
                "type": "date",
              }
            }
            value=""
          />,
        ],
        "className": " form__group",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeFunction": [Function],
            "className": "",
            "field": Object {
              "ariaLabel": "input field",
              "autoComplete": "off",
              "id": "inputFied",
              "inputType": "input",
              "placeholder": "Type a message...",
              "type": "text",
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeFunction": [Function],
            "className": "",
            "field": Object {
              "autoComplete": "off",
              "id": "textFied",
              "inputType": "textarea",
              "placeholder": "Type a message...",
              "type": "text",
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeFunction": [Function],
            "className": "",
            "field": Object {
              "id": "selectField",
              "inputType": "select",
              "selectOptions": Array [
                Object {
                  "text": "Example 1",
                  "value": "1",
                },
                Object {
                  "text": "Example 2",
                  "value": "2",
                },
              ],
              "type": "checkbox",
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeFunction": [Function],
            "className": "",
            "field": Object {
              "id": "datepicker",
              "inputType": "datepicker",
              "type": "date",
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
